### DISCUSSION TITLE
Custom directoy

### DISCUSSION THREAD

**Discussion by diderich** [YetiForceCompany/YetiForceCRM#16423] [How to] [answered]
I just created a test installation. 

To re-use the customization I made, I created a link `ln -s /opt/crm6/custom custom` in the root directory of the test installation to my production installation.

But this leads to the following error and me being unable to login.
```
PHP Fatal error:  Uncaught App\\Exceptions\\AppException: Sorry! Attempt to access restricted file. in /opt/crm6test/vtlib/Vtiger/Deprecated.php:75
Stack trace:
#0 /opt/crm6test/include/Loader.php(110): vtlib\\Deprecated::checkFileAccessForInclusion()
#1 /opt/crm6test/include/Loader.php(215): Vtiger_Loader::includeOnce()
#2 /opt/crm6test/modules/Vtiger/models/Module.php(205): Vtiger_Loader::autoLoad()
#3 /opt/crm6test/modules/Vtiger/models/Module.php(188): Vtiger_Module_Model::getInstanceFromModuleObject()
#4 /opt/crm6test/modules/Users/models/Privileges.php(104): Vtiger_Module_Model::getInstance()
#5 /opt/crm6test/modules/Vtiger/views/DashBoard.php(19): Users_Privileges_Model->hasModulePermission()
#6 /opt/crm6test/include/main/WebUI.php(185): Vtiger_DashBoard_View->checkPermission()
#7 /opt/crm6test/index.php(26): Vtiger_WebUI->process()
#8 /opt/crm6test/public_html/index.php(13): require('...')
#9 {main}\n  thrown in /opt/crm6test/vtlib/Vtiger/Deprecated.php on line 75
```
Is it correct that the `custom` directory must be an actual directory and and cannot be a link (or is it a bug)?

**Comment by mariuszkrzaczkowski** [comment, marked as answer]
This is an old vtiger mechanism, as far as I can remember there were difficulties with Symbolic Links working.
It would require testing, vtiger had problems loading files and paths, which is probably why they did such functions.
it was needed when composer and automatic file loading were not available,
I would gladly throw it away, but it requires a lot of testing and time to break something
